wsi_superpixel_guided_labeling/web_client/views/vue/components/ActiveLearningSetup/ActiveLearningInitialLabels.vue

```diff
@@ -246,18 +251,23 @@ export default Vue.extend({
             superpixelElement.values = data;
             this.debounceSaveAnnotations();
         },
-        updateRunningLabelCounts(boundaries, index, categoryIndex) {
+        updateRunningLabelCounts(boundaries, index, newLabel, oldLabel) {
             const elementValueIndex = boundaries ? index / 2 : index;
-            const currentCategoryIndices = this.categories[this.categoryIndex].indices[this.currentImageId] || [];
-            if (categoryIndex === 0) {
-                this.categories[this.categoryIndex].indices[this.currentImageId] = _.filter(currentCategoryIndices, (i) => i !== elementValueIndex);
-            } else {
-                currentCategoryIndices.push(elementValueIndex);
+            const currentCategoryIndices = this.categories[this.categoryIndex].indices[this.currentImageId] || new Set();
+            if (!currentCategoryIndices.size) {
                 this.categories[this.categoryIndex].indices[this.currentImageId] = currentCategoryIndices;
-                // Force computed properties to update
-                const newCategoryData = Object.assign({}, this.categories[this.categoryIndex]);
-                this.categories.splice(this.categoryIndex, 1, newCategoryData);
             }
+            if (newLabel === 0) {
+                currentCategoryIndices.delete(elementValueIndex);
+            } else if (oldLabel === 0) {
+                currentCategoryIndices.add(elementValueIndex);
+            } else {
+                this.categories[oldLabel - 1].indices[this.currentImageId].delete(elementValueIndex);
+                currentCategoryIndices.add(elementValueIndex);
```

Comment:
This is the new case. If there's a superpixel that's labeled category X, and I want to label it category Y, this updates both counts.